### PR TITLE
chore: Switch developerTestAccount & sandbox usages from `lib` to `api`

### PR DESCRIPTION
**Pull request by joe-yeager**
## Description and Context

- Switches to use `api/developerTestAccounts.ts` instead of `lib/developerTestAccounts.ts` because we are deprecating `lib/developerTestAccounts.ts` because it is a very thin wrapper and we can directly use `api/developerTestAccounts.ts`
- Switches usages of `lib/sandboxes.ts` to `api/sandboxHubs` and `api/sandboxSync`

Companion PR: https://github.com/HubSpot/hubspot-local-dev-lib/pull/178

## Who to Notify

@kemmerle @adamawang @camden11 @brandenrodgers

### CODE REVIEW

File: packages/cli/lib/developerTestAccounts.js [hunk 1]
```diff
@@ -32,31 +32,31 @@ const getHasDevTestAccounts = appDeveloperAccountConfig => {
...
+          limit,
+        })
+      );
+    } else {
+      throw new Error(
+        i18n('lib.developerTestAccount.create.failure.limit', {
+          accountName: accountConfig.name || accountId,
+          limit,
+        })
+      );
```
Comment (joe-yeager) [inline packages/cli/lib/developerTestAccounts.js:56]: This isn't related, but I just flipped the if statement and returned `null` up front to reduce a layer of nesting

**Review by adamawang — APPROVED** [review]
Looks good with the local-dev-lib PR 👍

**Comment by adamawang** [comment]
Should I look into doing something similar with the [sandboxes](https://github.com/HubSpot/hubspot-local-dev-lib/blob/main/lib/sandboxes.ts) file? Since that doesn't really add much either

**Comment by joe-yeager** [comment]
> Should I look into doing something similar with the [sandboxes](https://github.com/HubSpot/hubspot-local-dev-lib/blob/main/lib/sandboxes.ts) file? Since that doesn't really add much either

@adamawang Oh, good to know.  I'll just add it to these PRS since I'm already in here and we are trying to build up some breaking changes for a v2 of LDL

File: packages/cli/commands/sandbox/delete.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ const { trackCommandUsage } = require('../../lib/usageTracking');
 const { loadAndValidateOptions } = require('../../lib/validation');
 const { logError, debugError } = require('../../lib/errorHandlers/index');
 const { isSpecifiedError } = require('@hubspot/local-dev-lib/errors/index');
-const { deleteSandbox } = require('@hubspot/local-dev-lib/sandboxes');
+const { deleteSandbox } = require('@hubspot/local-dev-lib/api/sandboxHubs');
```
Comment (joe-yeager) [inline packages/cli/commands/sandbox/delete.js:13]: The call is on line 141 and the return value is being ignored, that's why there are no other changes here.

**Review by camden11 — APPROVED** [review]
LGTM. From what I can tell, this doesn't depend on any upcoming changes in LDL either so you could safely merge into main rather than next if you wanted to

**Comment by joe-yeager** [comment]
> LGTM. From what I can tell, this doesn't depend on any upcoming changes in LDL either so you could safely merge into main rather than next if you wanted to

I believe it's depending on the changes you made to the API calls returning the full promise

**Comment by camden11** [comment]
🤦 It absolutely is, i'm losing track of what's on `next` and what's not haha

**Review by brandenrodgers — APPROVED** [review]

**Pull request merged**